ocaml/xapi/xapi_ha.ml

```diff
@@ -834,27 +834,35 @@ let ha_stop_daemon __context localhost =
   let (_ : string) = call_script ha_stop_daemon [] in ()
 
 (** Emergency-mode API call to disarm localhost *)
-let emergency_ha_disable __context =
-  if Localdb.get Constants.ha_armed = "false"
-  then raise (Api_errors.Server_error(Api_errors.ha_not_enabled, []));
-
-  warn "Host.emergency_ha_disable: Disabling the HA subsystem on the local host only.";
-  Localdb.put Constants.ha_armed "false";
-
-  begin
-    try
-      ha_disarm_fencing __context ();
-    with Xha_error e ->
-      error "Host.emergency_ha_disable: ha_disarm_fencing failed with %s; continuing" (Xha_errno.to_string e)
-  end;
-  begin
-    try
-      ha_stop_daemon __context ();
-    with Xha_error e ->
-      error "Host.emergency_ha_disable: ha_stop_daemon failed with %s; continuing" (Xha_errno.to_string e)
-  end;
-  (* Might not be able to access the database to detach statefiles; however this isn't critical *)
-  ()
+let emergency_ha_disable __context soft =
+  let ha_armed = Localdb.get Constants.ha_armed in
+  if ha_armed = "false" then
+    if soft then () else raise (Api_errors.Server_error(Api_errors.ha_not_enabled, []))
+  else begin
+    warn "Host.emergency_ha_disable: Disabling the HA subsystem on the local host only.";
+    Localdb.put Constants.ha_armed "false";
```

Comment:
No particular reason except that the code been replaced also changed the config before the disabling action.

Nonetheless, after checking the places referencing this config variable, it seem that changing the config afterwards could be more conservative/safer (if there is any difference). 

As such, I've updated the code accordingly.